File: truth/backend/src/truth_service/models/industry.py
"""
The payloads of the industries, which are the vocabulary the register is filed under.

:date: 2026-09-08
"""
# ----- IMPORTS ----- #

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

# ----- CLASSES ----- #


class IndustryResponse(BaseModel):
    """
    One industry as every endpoint hands it over.

    The count is filled in only by the listing that was asked for it: counting the assumptions of every
    industry is a pass over the whole register, and most readers of this payload only want the name.
    """

    model_config = ConfigDict(populate_by_name=True)

    id: str = Field(description="Identifier of the industry")
    name: str = Field(description="Name of the industry")
    description: str = Field(default="", description="What the industry covers")
    creator: str = Field(default="", description="User that registered the industry")
    created_at: datetime | None = Field(default=None, description="UTC moment the industry was registered")
    assumption_count: int | None = Field(default=None, description="How many assumptions name this industry")


class IndustryCreateRequest(BaseModel):
    """
    What registering an industry asks for.
    """

    model_config = ConfigDict(populate_by_name=True)

    name: str = Field(min_length=1, description="Name of the industry")
    description: str = Field(default="", description="What the industry covers")
    creator: str = Field(default="", description="User registering the industry")

File: truth/frontend/src/models/industry.ts
/**
 * The payloads of the industries, which are the top of the navigation and the vocabulary assumptions belong to.
 */

/**
 * An industry as every endpoint of the API hands it over - an identifier and a name, and nothing else.
 *
 * The service holds a description and a creator as well, but never gives either of them back, so nothing in
 * the client may be written as though it had them.
 */
interface Industry {
  id: string
  name: string
}

/** What creating an industry asks for. */
interface IndustryDraft {
  name: string
  description: string
  creator: string
}

export type { Industry, IndustryDraft }
